### PR TITLE
[Model Monitoring] Feature Importance improvements

### PR DESCRIPTION
**Pull request by hawestra**
1. Split the train and test data when creating the RAI insights object and add a bit more logging. This is to help with the big data scenarios and address : [Bug 2708811](https://msdata.visualstudio.com/Vienna/_workitems/edit/2708811): Feature Importance takes >8 hours to run with 130k samples

Example run : https://ml.azure.com/experiments/id/f254e824-90b3-4c4a-bb95-7b3882e3a304/runs/amiable_oxygen_qrfylsbhb8?wsid=/subscriptions/5f08d643-1910-4a38-a7c7-84a39d4f42e0/resourceGroups/model-monitoring-bugbash-2023-05-09-rg/providers/Microsoft.MachineLearningServices/workspaces/momo-bugbash-2023-05-09-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

2. Fix error when the task type is not defined: https://ml.azure.com/experiments/id/41d84cfe-1b5b-4242-a8b1-7fe3413b17ed/runs/polite_arm_nvhwg4y7lv?wsid=/subscriptions/e0fd569c-e34a-4249-8c24-e8d723c7f054/resourcegroups/hawestra-rg/providers/Microsoft.MachineLearningServices/workspaces/hawestra-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

### CODE REVIEW

File: assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_importance.py [hunk 1]
```diff
@@ -256,8 +271,7 @@ def run(args):
             return
 
         baseline_df = baseline_df.toPandas()
-        task_type = args.task_type if args.task_type else determine_task_type(args.target_column, baseline_df)
```
Comment (yunjie-hub) [inline assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_importance.py:259]: You already have the None Check for the passed-in args parameter. but it still crashes, i believe the None is returned from determine_task_type(args.target_column, baseline_df)

File: assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_importance.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 )
 from shared_utilities.io_utils import try_read_mltable_in_spark_with_warning, save_spark_df_as_mltable, init_spark
 from shared_utilities import constants
-
+from sklearn.model_selection import train_test_split
```
Comment (yunjie-hub) [inline assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_importance.py:16]: When we do code change for the component, can we also add unit test and component e2e testing for the code.

File: assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_importance.py (hunk 1 above)
Comment (hawestra) [reply assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_importance.py:259]: right I added more datatypes / better checking for each kind of datatype, plus default to classification in the catch all scenario

File: assets/model_monitoring/components/tests/unit/test_compute_feature_importance.py [hunk 3]
```diff
@@ -60,18 +60,22 @@ def test_compute_categorical_features_integer(self, get_zipcode_data):
...
+        task_type = determine_task_type(None, "ISPROXYIP", get_fraud_data)
         assert task_type == "classification"
-
-        task_type = determine_task_type("zipcode", get_zipcode_data)
+        task_type = determine_task_type("invalid", "zipcode", get_zipcode_data)
+        assert task_type == "classification"
+        task_type = determine_task_type("Classification", "zipcode", get_zipcode_data)
         assert task_type == "classification"
 
     def test_determine_task_type_regression(self, get_fraud_data):
```
Comment (yunjie-hub) [inline assets/model_monitoring/components/tests/unit/test_compute_feature_importance.py:70]: Could you please also add the case to the overridden like, customer passed in regression while from infer type is classification

**Review by yunjie-hub — APPROVED** [review]

**Pull request merged**